harness/determined/experimental/client.py

```diff
@@ -233,7 +233,21 @@ def get_model_by_name(name: str) -> Model:
         name (str): The unique name of the model.
     """
     assert _determined is not None
-    matching_models = _determined.get_models(name=name)
+    return _determined.get_model(name)
+
+
+@_require_singleton
+def get_model_by_id(model_id: int) -> Model:
+    """
+    Get the :class:`~determined.experimental.client.Model` from the model registry
+    with the provided numeric id. If no model with that id is found in the registry,
+    an exception is raised.
+
+    Arguments:
+        model_id (int): The unique id of the model.
+    """
+    assert _determined is not None
+    matching_models = _determined.get_models(model_id=model_id)
     assert len(matching_models) > 0
     return matching_models[0]
```

Comment:
this logic seems suspect.  What if there are multiple matching models?

Reply:
maybe `assert len(matching_models) == 1` then?

Reply:
will do ✅